SMLHelper/Logger.cs

```diff
@@ -1,36 +1,80 @@
 ﻿namespace SMLHelper.V2
 {
     using System;
+    using System.IO;
+
+    internal enum LogLevel
+    {
+        Debug,
+        Warn,
+        Error,
+        Info
+    }
 
     internal static class Logger
     {
-        internal static void Log(string text)
+        internal static LogLevel LoggingLevel { get; private set; }
+
+        internal static void Initialize()
+        {
+            string configPath = "./QMods/Modding Helper/loglevel.txt";
+
+            if(!File.Exists(configPath))
+            {
+                File.AppendAllText(configPath, "Warn");
+                LoggingLevel = LogLevel.Warn;
+
+                return;
+            }
+
+            string level = File.ReadAllText(configPath);
+
+            try
+            {
+                LoggingLevel = (LogLevel)Enum.Parse(typeof(LogLevel), level);
+
+                Log($"Log level set to: LogLevel.{LoggingLevel.ToString()}.", LogLevel.Info);
+            }
+            catch (Exception)
+            {
+                LoggingLevel = LogLevel.Warn;
+
+                File.Delete(configPath);
+                File.AppendAllText(configPath, "Warn");
+
+                Log("Error reading log level configuration file. Defaulted to LogLevel.Warn.", LogLevel.Warn);
+            }
+        }
+
+        internal static void Log(string text, LogLevel level = LogLevel.Error)
         {
-            Console.WriteLine($"[SMLHelper] {text}");
+            if(level >= LoggingLevel)
+                Console.WriteLine($"[SMLHelper/{level.ToString()}] {text}");
```

Comment:
Maybe also check for invalid log ids before calling To String

Reply:
You really can't have invalid log ids.